packages/common/src/visualizations/BarChartDataTransformer.ts

```diff
@@ -25,6 +25,31 @@ export class BarChartDataTransformer<TBarChartLayout, TPieChartConfig> {
 
         const DEFAULT_X_AXIS_TYPE = 'category';
 
+        const series = transformedData?.seriesColumns.map(
+            (seriesColumn, index) => {
+                const seriesLabel = Object.values(display?.series || {}).find(
```

Comment:
this is where I'm a bit confused if I should do it this way @owlas 

We want to get the label, we know the series is an object where the key is a `reference` and the values include `yAxisIndex` 
Would you do this differently or potentially change the structure of series? 🤔